MCLOS - Add support for AI

PabstMirror opened the pull request:
For PR #10027

- Adds basic support for AI shooting MCLOS weapons
It's not very accurate and there is a lot of oscillation in the path, but that's probably pretty accurate for these types of missiles.

- Also adds a mock vic so this can be tested in vanilla arma
Disabled by config macro, but could leave in as scope=1 if that's ok? (makes testing not require restart)

TheCandianVendingMachine commented on addons/missileguidance/functions/fnc_seekerType_MCLOS.sqf:
```diff
@@ -16,16 +16,39 @@
+    private _distance = _launchPos vectorDistance _projectilePos;
+    private _projectedPath = _launchPos vectorAdd (_launchVec vectorMultiply _distance);
+
+    private _source = _projectile getVariable [QGVAR(source), _projectile];
```
is this unused?

TheCandianVendingMachine commented on addons/missileguidance/functions/fnc_seekerType_MCLOS.sqf:
```diff
@@ -16,16 +16,39 @@
+    private _source = _projectile getVariable [QGVAR(source), _projectile];
+    private _hOffset = (_projectilePos # 2) - (_projectedPath # 2);
+    private _vv = (velocity _projectile) # 2;
+    _mclosInput = switch (true) do {
```
this doesn't support side-to-side travel

TheCandianVendingMachine commented on addons/missileguidance/functions/fnc_seekerType_MCLOS.sqf:
```diff
@@ -16,16 +16,39 @@
+    if (!(alive _shooter && {alive _gunner} && {(vehicle _gunner) == _shooter} && {_gunner call EFUNC(common,isAwake)})) exitWith {};
+
+    private _distance = _launchPos vectorDistance _projectilePos;
+    private _projectedPath = _launchPos vectorAdd (_launchVec vectorMultiply _distance);
```
this only aims for the point the ai was aiming when they fired. if the target moves, this inherently misses. i think we could do something similar to SACLOS where we figure out the position the gunner is looking but as SACLOS shows that isn't easy. this is good for now, just wanted to note this

PabstMirror commented:
I moved the `hasMCLOSControl` to the weapon itself
otherwise everyone in the vic had access
could do a turret path config, but I didn't see any downside to moving it to the weapon,
also makes it easier to add/remove weapons via script

also dropped the click volume down as it's a bit much when you are making constant adjustments
